### PR TITLE
adding revocation errors

### PR DESCRIPTION
**Pull request by seanturner**
@bifurcation This definitely needs to be reviewed.

### CODE REVIEW

**Comment by richsalz** [comment]
This looks good to me.

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -2231,7 +2232,9 @@ The server MUST also consider a revocation request valid if it is signed with
 the private key corresponding to the public key in the certificate.
 
 If the revocation succeeds, the server responds with status code 200 (OK).  If
-the revocation fails, the server returns an error.
+the revocation fails, the server returns an error.  For example, if the certificate
+has already been revoked the servers returns an error response with status code 400
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: nit: "the servers returns" -> "the server returns"

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -547,6 +547,7 @@ in the "type" field (within the "urn:ietf:params:acme:error:" namespace):
 | Type                    | Description                                                                    |
 |:------------------------|:-------------------------------------------------------------------------------|
 | accountDoesNotExist     | The request specified an account that does not exist                           |
+| alreadyRevoked          | The certificate has already been revoked                                       |
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: I think this description would might match the others a little more closely if it were phrased along the lines of "The request specified a certificate to be revoked that has already been revoked"

**Comment by uhhhh2** [comment]
I like the idea of a special compound error type to indicate that the
subproblems field is used.

**Pull request merged**